### PR TITLE
feat: templates ui ux updates

### LINKED ISSUE #25402: Templates UI enhancements
Improve the templates UI based on recommendations from design team [here](https://www.notion.so/appsmith/Templates-tab-on-the-platform-design-review-9a688315329d4776a879b0bfe9252f56?pvs=4)

- [x] (point 2) Removed `Filters` heading
- [x] (point 3) In `Add a page from template` flow remove  `+` icon
- [x] (point 6) Add scrollbar in appropriate view ports
- [x] (point 9) Scrollbar in templates pages list

### PR DESCRIPTION
**Pull request by rahulbarwal**
## Description
1. imports only required stuff from object explorer
2. updates code to utilize helper functions.
#### PR fixes following issue(s)
Fixes #25402
- Bug fix (non-breaking change which fixes an issue)
- [x] Cypress
>
>
#### Dev activity
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

#### QA activity:
- [ ] [Speedbreak features](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#speedbreakers-) have been covered
- [ ] Test plan covers all impacted features and [areas of interest](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#areas-of-interest-)
- [ ] Test plan has been peer reviewed by project stakeholders and other QA members
- [ ] Manually tested functionality on DP
- [ ] We had an implementation alignment call with stakeholders post QA Round 2
- [ ] Cypress test cases have been added and approved by SDET/manual QA
- [ ] Added `Test Plan Approved` label after Cypress tests were reviewed
- [ ] Added `Test Plan Approved` label after JUnit tests were reviewed

### CODE REVIEW

File: app/client/cypress/e2e/Regression/ClientSide/Templates/Fork_Template_Existing_app_spec.js [hunk 1]
```diff
@@ -21,22 +25,18 @@ describe(
...
-      _.agHelper.AssertElementExist(_.templates.locators._forkApp);
-      cy.get(template.templateDialogBox).should("be.visible");
-      cy.wait(4000);
-      cy.xpath(
-        "//h1[text()='Meeting Scheduler']/parent::div//button[contains(@class, 't--fork-template')]",
-      )
+      entityExplorer.AddNewPage("Add page from template");
+      agHelper.Sleep(5000);
+      agHelper.AssertElementExist(template.templateDialogBox);
+      agHelper.AssertElementExist(templates.locators._templateCard);
```
Comment (Aishwarya-U-R) [inline app/client/cypress/e2e/Regression/ClientSide/Templates/Fork_Template_Existing_app_spec.js:31]: We have `AssertElementVisible` too.

Comment (rahulbarwal) [reply app/client/cypress/e2e/Regression/ClientSide/Templates/Fork_Template_Existing_app_spec.js:31]: Thanks for pointing out. Used here and 2 more places.

**Review by Aishwarya-U-R — COMMENTED** [review]
Approving Cypress changes.

**Review by akash-codemonk — DISMISSED** [review]

**Comment by shwetha-ramesh** [comment]
@rahulbarwal Tested this DP and observed this - when a user clicks on any Template from 'Add page from template' when user scrolls the modal, under `Similar templates` `+` icon is visible.
image

**Comment by rahulbarwal** [comment]
@shwetha-ramesh Thanks for pointing that out. I had missed that. Added no, DP: https://appsmith-26dwialpd-get-appsmith.vercel.app/

**Comment by shwetha-ramesh** [comment]
@rahulbarwal QA Done

**Review by abm17 — APPROVED** [review]

**Pull request merged**